File: src/app/models/bon-sortie.ts
import {DetailSortie} from "./detail-sortie";
import {Utilisateur} from "./utilisateur";

export interface BonSortie {
  id: number;
  motif: string;
  date_sortie: Date;
  utilisateur: Utilisateur;
  detailsSorties: DetailSortie[];
}

File: Test-SB/src/main/java/com/test/entities/BonSortie.java
package com.test.entities;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDate;

@Entity
@Table(name = "bon_sorties")
@Data
@AllArgsConstructor
@NoArgsConstructor
public class BonSortie {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private int Id;

    @Column(name = "motif")
    private String motif;

    @Column(name = "date_sortie", nullable = false)
    private LocalDate dateSortie;

    @ManyToOne
    @JoinColumn(name = "utilisateur_id", nullable = false)
    private Utilisateur utilisateur;
}
